lib_standard_app/debug.c

```diff
@@ -59,13 +60,42 @@ WEAK void __attribute__((noreturn)) debug_display_throw_error(int exception)
 
 #ifdef HAVE_NBGL
     nbgl_useCaseChoice(
-        &C_round_warning_64px, "App error", errordata, "Exit app", "Exit app", review_choice);
+        &C_round_warning_64px, "App error", errordata, "Exit app", "Exit app", exit_app);
 #endif
 
     // Block until the user approve and the app is quit
     while (1) {
         io_seproxyhal_io_heartbeat();
     }
 }
+#endif
+
+#ifdef HAVE_LEDGER_ASSERT
+#define ASSERT_ERROR_CODE 0xBAAD
+void ledger_assert(char *message)
```

Comment:
With a function like that, the message string will always be embedded in the app binary, even when unused (when PRINTF are disabled).

I think we should go for an ASSERT made of a macro calling different functions in the lines of what's done here: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/__assert.h

- the ASSERT macro check for a result, so that it can be called unconditionally in the app code.
- the ASSERT macro can display a message or not, but doesn't hold it in the app binary if it's not being displayed
- the ASSERT macro can display file name and line, but this should be disabled by default as this use a lot of flash
- the ASSERT macro can access both the PC and the LR (instead of just the PC which is called LR here as it's a macro and not a function).